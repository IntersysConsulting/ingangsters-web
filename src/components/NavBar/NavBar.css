.navbar,
.navbar-bg-gray {
  background-color: #55565a;
  z-index: 100;
}

.btn,
.btn-searchBar {
  height: 30px;
  color: white;
}
