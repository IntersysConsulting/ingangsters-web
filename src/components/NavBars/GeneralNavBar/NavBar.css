.navbar,
.navbar-bg-gray {
  background-color: var(--dark-grey);
  z-index: 100;
  border-bottom: 3px solid var(--light-blue);
  position: relative;
  /* position: fixed;
  width: 100%;
  top: 0px; */
}

.navbar-responsive .iconButtonWrapper {
  display: flex;
  flex-direction: row;
}

.navbar-responsive {
  display: flex;
  flex-direction: row;
}

.navbar-responsive .cbButton,
.cbSeparator {
  display: none;
}

.navbar-responsive .buttonIcon {
  font-size: 2.3em;
}

.navbar-responsive .buttonLabel {
  margin-bottom: 0;
}

@media screen and (max-width: 550px) {
  .navbar {
    width: 100%;
    padding: 0;
    margin: 0;
    border-bottom: none;
  }
  .navbar-responsive {
    flex-direction: column;
    overflow: hidden;
    height: 90px;
    transition: height 0.3s ease-in;
    width: 100%;
    padding: 0 !important;
  }

  .navbar-responsive .navbar-brand {
    width: 100%;
    margin-top: 2px;
    margin-bottom: 3px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-right: 0 !important;
    border-bottom: 3px solid var(--light-blue);
  }

  .navbar-responsive .iconButtonWrapper {
    width: 100%;
    justify-content: space-around;
    margin-top: 0;
  }

  .navbar-responsive .iconButtonWrapper .verticalSeparator {
    width: 2px;
    background-color: var(--light-blue);
    height: 80%;
  }

  .navbar-responsive.active {
    height: 200px;
    transition: all 0.3s ease-out;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }

  .navbar-responsive .buttonLabel {
    display: none;
  }

  .navbar-responsive .buttonIcon {
    font-size: 3.4em;
  }

  .navbar-responsive .cbButton {
    display: inline;
    position: absolute;
    right: 10px;
    cursor: pointer;
    top: 20px;
  }
}
