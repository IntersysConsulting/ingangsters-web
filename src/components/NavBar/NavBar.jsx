import React from "react";
import NavBar from "react-bootstrap/Navbar";
import NavBarBrand from "react-bootstrap/NavbarBrand";
import SearchBar from "./SearchBar/SearchBar";
import IconButton from "./IconButton/IconButton";
import { MdPerson } from "react-icons/md";
import { MdShoppingCart } from "react-icons/md";
import { setAccountButtonClicked } from "../../actions/creators/navBar";
import { showShadow } from "../../actions/creators/shadow";
import { hideShadow } from "../../actions/creators/shadow";
import { connect } from "react-redux";
import "./NavBar.css";

const NavBarComponent = ({ shadowActive }) => (
  <NavBar bg="bg-gray" style={{ display: "flex", flexDirection: "row" }}>
    <NavBarBrand>
      <img
        alt=""
        src="/assets/logo.png"
        width="75"
        height="75"
        className="mx-auto"
      />
    </NavBarBrand>
    <div style={{ flex: 1 }} />
    <SearchBar />
    <div style={{ flex: 1 }} />
    <div style={{ display: "flex", flexDirection: "row" }}>
      <IconButton
        Icon={MdPerson}
        Label="Account"
        ClickEvent={() => {
          console.log("clicked account");
        }}
        Dispatch={shadowActive ? hideShadow() : showShadow()}
      />
      <IconButton
        Icon={MdShoppingCart}
        Label="Cart"
        Dispatch={{ type: "NAAN" }}
      />
    </div>
  </NavBar>
);

const mapStateToProps = state => ({
  shadowActive: state.shadow.active
});
export default connect(
  mapStateToProps,
  {}
)(NavBarComponent);
