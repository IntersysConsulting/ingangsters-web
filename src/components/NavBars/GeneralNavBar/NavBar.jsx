import React from "react";
import NavBar from "react-bootstrap/Navbar";
import NavBarBrand from "react-bootstrap/NavbarBrand";
import SearchBar from "./SearchBar/SearchBar";
import IconButton from "./IconButton/IconButton";
import { MdPerson } from "react-icons/md";
import { MdShoppingCart } from "react-icons/md";
import { toggleMenuResponsive } from "../../../actions/creators/navBar";
import { showShadow } from "../../../actions/creators/shadow";
import { hideShadow } from "../../../actions/creators/shadow";
import { connect } from "react-redux";
import { MdMenu } from "react-icons/md";
import { FaRegWindowClose } from "react-icons/fa";
import "../../../css/colors.css";
import "./NavBar.css";
import { hideModal } from "../../actions/creators/modal";
import { showModal } from "../../actions/creators/modal";
import Modal from "../Modal/Modal";

const NavBarComponent = ({
  shadowActive,
  responsiveMenuActive,
  toggleMenuResponsive,
  _dispatch
}) => {
  var navBarClassList = ["navbar-responsive"];
  let ToggleButton;
  if (responsiveMenuActive) {
    navBarClassList.push("active");
    ToggleButton = FaRegWindowClose;
  } else {
    ToggleButton = MdMenu;
  }

  return (
<<<<<<< development:src/components/NavBars/GeneralNavBar/NavBar.jsx
    <NavBar bg="bg-gray" className={navBarClassList.join(" ")}>
      <NavBarBrand>
        <img
          alt=""
          src="/assets/logo.png"
          width="75"
          height="75"
          className="mx-auto"
        />
        <ToggleButton
          size="2.4em"
          color="#ffffff"
          className="cbButton"
          onClick={() => {
            toggleMenuResponsive();
          }}
        />
      </NavBarBrand>
      <div className="flexSeparator" style={{ flex: 1 }} />
      <SearchBar />
      <div className="flexSeparator" style={{ flex: 1 }} />
      <div className="iconButtonWrapper">
        <IconButton
          Icon={MdPerson}
          Label="Account"
          ClickEvent={() => {
            if (shadowActive) _dispatch(hideShadow());
            else _dispatch(showShadow());
          }}
        />
        <div className="verticalSeparator" />
        <IconButton Icon={MdShoppingCart} Label="Cart" />
      </div>
    </NavBar>
=======
    <div>
      <NavBar bg="bg-gray" className={navBarClassList.join(" ")}>
        <NavBarBrand>
          <img
            alt=""
            src="/assets/logo.png"
            width="75"
            height="75"
            className="mx-auto"
          />
          <ToggleButton
            size="2.4em"
            color="#ffffff"
            className="cbButton"
            onClick={() => {
              toggleMenuResponsive();
            }}
          />
        </NavBarBrand>
        <div className="flexSeparator" style={{ flex: 1 }} />
        <SearchBar />
        <div className="flexSeparator" style={{ flex: 1 }} />
        <div className="iconButtonWrapper">
          <IconButton
            Icon={MdPerson}
            Label="Account"
            ClickEvent={() => {
              //console.log("clicked account");
              if (shadowActive) {
                _dispatch(hideModal());
                _dispatch(hideShadow());
              } else {
                _dispatch(showModal());
                _dispatch(showShadow());
              }
            }}
          />
          <div className="verticalSeparator" />

          <IconButton Icon={MdShoppingCart} Label="Cart" />
        </div>
      </NavBar>
      <Modal />
    </div>
>>>>>>> Responsive modal and the respective labels appears:src/components/NavBar/NavBar.jsx
  );
};

const mapStateToProps = state => ({
  shadowActive: state.shadow.active,
  responsiveMenuActive: state.navBar.responsiveMenuActive
});

const mapDispatchToProps = dispatch => ({
  toggleMenuResponsive() {
    dispatch(toggleMenuResponsive());
  },
  _dispatch(action) {
    dispatch(action);
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavBarComponent);
