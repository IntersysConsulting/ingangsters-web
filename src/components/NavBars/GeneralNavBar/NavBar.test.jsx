import React from "react";
import { shallow } from "enzyme";
import { NavBarComponent } from "./NavBar";

describe("General Navbar", () => {
  it("renders correctly", () => {
    const shallowRender = shallow(
      <NavBarComponent
        responsiveMenuActive={false}
        toggleMenuResponsive={() => {
          console.log("Toggle menu responsive");
        }}
      />
    );

    expect(shallowRender).toMatchSnapshot();
  });

  it("renders cheeseburguer button", () => {
    const wrapper = shallow(
      <NavBarComponent
        responsiveMenuActive={false}
        toggleMenuResponsive={() => {
          console.log("Toggle menu responsive");
        }}
      />
    );

    const burguerButtonList = wrapper.find("MdMenu");
    expect(burguerButtonList.length).toBe(1);
  });

  it("hides buttons on responsitve menu inactive", () => {
    const wrapper = shallow(
      <NavBarComponent
        responsiveMenuActive={false}
        toggleMenuResponsive={() => {
          console.log("Toggle menu responsive");
        }}
      />
    );

    const navBarActiveList = wrapper.find(".navbar-responsive.active");
    expect(navBarActiveList.length).toBe(0);
  });

  it("shows buttons on responsitve menu active", () => {
    const wrapper = shallow(
      <NavBarComponent
        responsiveMenuActive={true}
        toggleMenuResponsive={() => {
          console.log("Toggle menu responsive");
        }}
      />
    );

    const navBarActiveList = wrapper.find(".navbar-responsive.active");
    expect(navBarActiveList.length).toBe(1);
  });

  it("renders close button", () => {
    const wrapper = shallow(
      <NavBarComponent
        responsiveMenuActive={true}
        toggleMenuResponsive={() => {
          console.log("Toggle menu responsive");
        }}
      />
    );

    const closeButtonList = wrapper.find("FaRegWindowClose");
    expect(closeButtonList.length).toBe(1);
  });
});
