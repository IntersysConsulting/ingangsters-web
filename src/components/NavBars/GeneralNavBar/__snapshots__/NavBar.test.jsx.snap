// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Navbar renders correctly 1`] = `
<div>
  <Navbar
    bg="bg-gray"
    className="navbar-responsive"
    collapseOnSelect={false}
    expand={true}
    variant="light"
  >
    <NavbarBrand>
      <Link
        to="/"
      >
        <img
          alt=""
          className="mx-auto"
          height="75"
          src="/assets/logo.png"
          width="75"
        />
      </Link>
      <MdMenu
        className="cbButton"
        color="#ffffff"
        onClick={[Function]}
        size="2.4em"
      />
    </NavbarBrand>
    <div
      className="flexSeparator"
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Connect(withRouter(SearchBar)) />
    <div
      className="flexSeparator"
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <div
      className="iconButtonWrapper"
    >
      <Connect(ModalSchema) />
      <div
        className="verticalSeparator"
      />
      <IconButton
        Icon={[Function]}
        Label="Cart"
      />
    </div>
  </Navbar>
</div>
`;
